radix_sort.py

```diff
@@ -0,0 +1,73 @@
+def radix_sort(array):
+    """a sorting algorithm that looks through an array of integers,
+    and sorts them according to their base ten index"""
+    if len(array) > 0:
+        tens_place = 1
+        index_num = 0
+        max_val = max(array)
+        buckets = [[] for x in range(10)]
+        while max_val > tens_place:
+            for val in array:
+                index_num = val % (tens_place * 10) / tens_place
+                buckets[index_num].append(val)
+
+            transfer_list = []
+            for bucket in buckets:
+                for i in range(len(bucket)):
+                    transfer_list.append(bucket.pop(0))
```

Comment:
there is a more efficient way in Python to extend a list by adding another list in its entirety to it.  YOu should use that here rather than iterating through the sub-list items.